src/halide_ir.fbs

```diff
@@ -0,0 +1,692 @@
+    var: string;
+    for_type: ForType;
+    device_api: DeviceAPI;
+    dim_type: DimType;
+}
+
+enum LoopAlignStrategy: ubyte {
+    AlignStart,
+    AlignEnd,
+    NoAlign,
+    Auto,
+}
+
+table FuseLoopLevel {
+    fuse_level: LoopLevel;
+    align_dimension_names: [string];
+    align_strategies: [LoopAlignStrategy];
+}
+
+table FusedPair {
+    func_1: string;
+    func_2: string;
+    stage_1: int32;
+    stage_2: int32;
+    var_name: string;
+}
+
+table StageSchedule {
+    rvars: [ReductionVariable];
+    splits: [Split];
+    dims: [Dim];
+    prefetches: [PrefetchDirective];
+    fuse_level: FuseLoopLevel;
+    fused_pairs: [FusedPair];
+    touched: bool = false;
+    allow_race_conditions: bool = false;
+    atomic: bool = false;
+    override_atomic_associativity_test: bool = false;
+}
+
+table Definition {
+    is_init: bool;
+    predicate: Expr;
+    values: [Expr];
+    args: [Expr];
+    stage_schedule: StageSchedule;
+    specializations: [Specialization];
+    source_location: string;
+}
+
+table FuncMapping {
+    name: string;
+    index: int32;
+}
+
+enum NameMangling: ubyte {
+    Default,
+    C,
+    CPlusPlus,
+}
+
+table BufferConstraint {
+    min: Expr;
+    extent: Expr;
+    stride: Expr;
+    min_estimate: Expr;
+    extent_estimate: Expr;
+}
+
+table Parameter {
+    defined: bool;
+    is_buffer: bool;
+    type: Type;
+    dimensions: int32;
+    name: string;
+    data: uint64;
+    scalar_default: Expr;
+    scalar_min: Expr;
+    scalar_max: Expr;
+    scalar_estimate: Expr;
+}
+
+enum ExternFuncArgumentType: ubyte {
+    UndefinedArg,
+    FuncArg,
+    BufferArg,
+    ExprArg,
+    ImageParamArg,
+}
+
+table ExternFuncArgument {
+    arg_type: ExternFuncArgumentType;
+    func_index: int32;
+    buffer_name: string;
+    expr: Expr;
+    image_param_name: string;
+}
+
+table BufferDimension {
+    min: int32;
+    extent: int32;
+    stride: int32;
+}
+
+table Buffer {
+    defined: bool;
+    name: string;
+    type: Type;
+    dimensions: int32;
+    dims: [BufferDimension];
+    data: [uint8];
+}
+
+// Halide::internal::Function
+table Func {
+    name: string;
+    origin_name: string;
+    output_types: [Type];
+    required_types: [Type];
+    required_dims: int32;
+    args: [string];
+    func_schedule: FuncSchedule;
+    init_def: Definition;
+    updates: [Definition];
+    debug_file: string;
+    output_buffers_names: [string];
+    extern_arguments: [ExternFuncArgument];
+    extern_function_name: string;
+    extern_mangling: NameMangling;
+    extern_function_device_api: DeviceAPI;
+    extern_proxy_expr: Expr;
+    trace_loads: bool = false;
+    trace_stores: bool = false;
+    trace_realizations: bool = false;
+    trace_tags: [string];
+    frozen: bool = false;
+}
+
+table Pipeline {
+    funcs: [Func];
+    output_names: [string];
+    requirements: [Stmt];
+    func_names_in_order: [string];
+    parameters: [Parameter];
+    buffers: [Buffer];
+    // trace_pipeline: bool;
```

Comment:
As always, commented out code should be removed or have a comment, even for a draft PR :-)